app/src/pages/inside/logsPage/modals/copySendDefectModal/copySendDefectModal.jsx

```diff
@@ -0,0 +1,134 @@
+import { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { injectIntl, defineMessages } from 'react-intl';
+import { ModalLayout, withModal } from 'components/main/modal';
+import { activeProjectSelector } from 'controllers/user';
+import { showNotification, NOTIFICATION_TYPES } from 'controllers/notification';
+import { COMMON_LOCALE_KEYS } from 'common/constants/localization';
+import { fetch } from 'common/utils';
+import { URLS } from 'common/urls';
+
+const MESSAGE_TYPES = {
+  button: 'button',
+  header: 'header',
+  content: 'content',
+};
+
+const messages = defineMessages({
+  headerCopyTitle: {
+    id: 'CopySendDefectModal.headerCopyTitle',
+    defaultMessage: 'Receive previous result',
+  },
+  headerSendTitle: {
+    id: 'CopySendDefectModal.headerSendTitle',
+    defaultMessage: 'Send defect to the last item',
+  },
+  buttonCopyTitle: {
+    id: 'CopySendDefectModal.buttonCopyTitle',
+    defaultMessage: 'Receive',
+  },
+  buttonSendTitle: {
+    id: 'CopySendDefectModal.buttonSendTitle',
+    defaultMessage: 'Send',
+  },
+  contentCopyTitle: {
+    id: 'CopySendDefectModal.contentCopyTitle',
+    defaultMessage: 'Are you sure to receive defect data from previous failed item?',
+  },
+  contentSendTitle: {
+    id: 'CopySendDefectModal.contentSendTitle',
+    defaultMessage: 'Are you sure to send defect data to the last item?',
+  },
+  successMessage: {
+    id: 'CopySendDefectModal.successMessage',
+    defaultMessage: 'Defects have been updated',
+  },
+});
+
+@withModal('copySendDefectModal')
+@injectIntl
+@connect(
+  (state) => ({
+    activeProject: activeProjectSelector(state),
+  }),
+  {
+    showNotification,
+  },
+)
+export class CopySendDefectModal extends Component {
+  static propTypes = {
+    activeProject: PropTypes.string.isRequired,
+    showNotification: PropTypes.func.isRequired,
+    intl: PropTypes.object.isRequired,
+    data: PropTypes.shape({
+      lastHistoryItem: PropTypes.object,
+      itemForCopy: PropTypes.object,
+      isCopy: PropTypes.bool,
+      fetchFunc: PropTypes.func,
+    }).isRequired,
+  };
+
+  onInclude = (closeModal) => {
+    const {
+      intl,
+      activeProject,
+      data: { lastHistoryItem, itemForCopy, fetchFunc },
+    } = this.props;
+    const issues = [
+      {
+        issue: {
+          ...itemForCopy.issue,
+          externalSystemIssues: itemForCopy.issue.externalSystemIssues || [],
+        },
+        test_item_id: lastHistoryItem.id,
+      },
+    ];
+
+    fetch(URLS.testItem(activeProject, ''), {
```

Comment:
It's better to rewrite url function like this:
`testItem: (activeProject, id = '') => //...`
and then usage in this case will be:
`URLS.testItem(activeProject)`

Reply:
Sorry for my inattention. Fixed